src/link/Coff.zig

```diff
@@ -2597,25 +2579,66 @@ fn writeImportTables(coff: *Coff) !void {
     if (coff.idata_section_index == null) return;
     if (!coff.imports_count_dirty) return;
 
+    log.debug("writing import tables", .{});
+
     const gpa = coff.base.comp.gpa;
 
     const ext = ".dll";
     const header = &coff.sections.items(.header)[coff.idata_section_index.?];
 
+    // Allocate Imports within IAT.
+    // We assume IAT will come first thus we can allocate before calculating the
+    // total size of the section.
+    const Iat = struct {
+        rva: u32,
+        imports: std.ArrayList(Import.Index),
+    };
+    var iats = std.AutoArrayHashMap(u32, Iat).init(gpa);
+    defer {
+        for (iats.values()) |*iat| {
+            iat.imports.deinit();
+        }
+        iats.deinit();
+    }
+    for (coff.imports.items, 0..) |*import, i| {
+        const gop = try iats.getOrPut(import.lib_name);
+        if (!gop.found_existing) {
+            gop.value_ptr.* = .{
+                .rva = 0,
+                .imports = std.ArrayList(Import.Index).init(gpa),
+            };
+        }
+        const iat = gop.value_ptr;
+        import.rva = @as(u32, @intCast(iat.imports.items.len)) * @sizeOf(u64);
+        try iat.imports.append(@intCast(i));
+    }
```

Comment:
I am not convinced that you need to create a nested array here. At worst it should be an in-place sort.

Reply:
Mhm, I like this suggestion. Let's see what I can come up with.